Mount sudo check

NathanTP opened the pull request:
Marshal should work on systems without sudo, but using guestmount to move files around (especially for large images and lots of mounts) is slow. This PR auto-detects if the user has passwordless sudo available, and uses 'sudo mount -o loop' whenever possible to speed up build times.

Testing:
- full_test.sh passed (module known bugs fixed in #70) with sudo
- disk and nodisk tested with guestmount for test/flist.sh without sudo

a0u commented on wlutil/wlutil.py:
```diff
@@ -201,21 +201,35 @@ def waitpid(pid):
+    sudoCmd = "sudo"
+    @contextmanager
+    def mountImg(imgPath, mntPath):
+        run("sudo mount -o loop " + imgPath + " " + mntPath, shell=True)
```
All `run()` calls that might use `sudo` should not be a shell command if possible, in order to prevent any opportunities to inject arbitrary mount options through unescaped `imgPath` or `mntPath` strings.  It could be disastrous if either of those are ever controllable by user input.  (An attacker could mount over `/etc` and replace `/etc/passwd`.)

For a list of what could go wrong with `setuid` executables, this exchange about Calibre security vulnerabilities is a classic example: https://bugs.launchpad.net/calibre/+bug/885027

NathanTP commented:
I updated the mount commands (except cpio creation) to use non-shell run commands. This is slightly safer in some contexts. The cpio creation must be a shell command because the cpio program requires shell redirection and stuff. It does not consume any user-provided strings so it should be fine.

I also added a bit of a note to the docs/readme to highlight the security implications.

a0u commented on wlutil/build.py:
```diff
@@ -216,7 +216,7 @@ def makeInitramfs(srcs, cpioDir, includeDevNodes=False):
     for src in srcs:
         dst = cpioDir / (src.name + '.cpio')
-        run("find -print0 | cpio --owner root:root --null -ov --format=newc > " + str(dst), shell=True, cwd=src)
+        run(sudoCmd + " sh -c 'find -print0 | cpio --owner root:root --null -ov --format=newc' > " + str(dst), shell=True, cwd=src)
```
If `str(dst)` escapes all shell metacharacters, then this is fine as is.

NathanTP commented:
Factored cpio creation out to wlutil/wlutil.py:toCpio(). Now avoids shell mode completely.
